mantidproject/mantid
# Windows: Add support for Python 3 build

**Description of work.**

This adds support for building against Python 3 on Windows. Python 3.8 has been added to the thirdparty bundle. Note that this does not switch the default build to Python 3. This will be done later once all other OSes fully support this.

⚠️ Please make sure mantidproject/thirdparty-msvc2015#50 is merged when this pull request is merged. ⚠️ 

**To test:**

Run cmake as usual but with the `WITH_PYTHON3` switch enabled.
Try out MantidPlot and Workbench and check that things *look* okay.

Fixes #16578. 

*This does not require release notes* because **it does not do anything visibly for users yet.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
